table/tables/tables.go

```diff
@@ -113,9 +121,39 @@ func TableFromMeta(alloc autoid.Allocator, tblInfo *model.TableInfo) (table.Tabl
 	}
 
 	t.meta = tblInfo
+	partitionExpr, err := generatePartitionExpr(tblInfo)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	t.partitionExpr = partitionExpr
 	return t, nil
 }
 
+func generatePartitionExpr(tblInfo *model.TableInfo) ([]expression.Expression, error) {
+	pi := tblInfo.GetPartitionInfo()
+	if pi == nil {
+		return nil, nil
+	}
+	// TODO: Only partition by range is supported currently.
+	if pi.Type != model.PartitionTypeRange {
+		return nil, nil
```

Comment:
maybe throw a real error rather than `nil` ?

Reply:
From the user's point of view, we don't support it does not not means it's an error, is that right?

Reply:
Keep upgrading compatibility. @zhexuany 
We ignore table partition now, and user do not assume we support that.
If one day, when the user upgrade to a new version of TiDB and find it throw an error, he would feel, what the f**k ?